.github/workflows/linting.yaml

```
@@ -0,0 +1,61 @@
+name: code-style
+
+on:
+  push:
+    branches: "*"
+  pull_request:
+    branches: main
+
+
+
+jobs:
+
+  lint-black:
+    name: black-formatter
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.8
+      - name: Install Black
+        run: |
+          python -m pip install black
+      - name: Black Code Formatter
+        run: |
+          black --check . -v
+  # lint-flake8:
+  #   name: flake8
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - name: Checkout
+  #       uses: actions/checkout@v1
+  #     - name: Set up Python 3.8
+  #       uses: actions/setup-python@v1
+  #       with:
+  #         python-version: 3.8
+  #     - name: Install Flake8
+  #       run: |
+  #         python -m pip install flake8
+  #     - name: Flake8 check
+  #       run: |
+  #         flake8 .
+  # we should discuss if we want to enable this?
+  # lint-isort:
+  #   name: isort
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - name: Checkout
+  #       uses: actions/checkout@v1
+  #     - name: Set up Python 3.8
+  #       uses: actions/setup-python@v1
+  #       with:
+  #         python-version: 3.8
+  #     - name: Install isort
+  #       run: |
+  #         python -m pip install isort
+  #     - name: isort check
+  #       run: |
+  #         isort --check-only .
```

I would replace the black check with the pre-commit check 

```suggestion
name: code-style

on:
    push:
        branches: "*"
    pull_request:
        branches: master

jobs:
    pre-commit:
        name: pre-commit
        runs-on: ubuntu-latest

        steps:
            - name: checkout
              uses: actions/checkout@v2

            - name: set up python
              uses: actions/setup-python@v2
              with:
                  python-version: 3.8

            - name: Run pre-commit
              uses: pre-commit/action@v2.0.0
```

---

Thanks for the suggestion. This would just run whatever we have set in our `.pre-commit-config.yaml`? I am just trying to understand the difference in a bit more detail.

---

What would be the pros and cons of this @andersy005 ? Just a more consistent/compact setup?

---

>  Just a more consistent/compact setup?

Yes... The pre-commit hook check GitHub action guarantees that the settings and versions used by pre-commit hooks in a local setup are the same as the ones used in the CI.

---

Also, you get to check everything and not just the Python files...

---

Awesome. I just pushed the changes. Do you think this is good?

---

Do you want the CI to fail when codecov doesn't pass? If not, you can make codecov checks to be optional by adding `codecov.yml` at the top level:


```yaml
codecov:
  require_ci_to_pass: no
  max_report_age: off

comment: false

coverage:
  precision: 2
  round: down
  status:
    project:
      default:
        target: 95
        informational: true
    patch: off
    changes: off
```

---

Wow. How did you read my mind? Hahaha I was literally going to ask that